core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -81,85 +90,103 @@ final class JdbcUtil {
           + TABLE_NAME
           + ")"
           + ")";
-  static final String GET_TABLE_SQL =
+  static final String UPDATE_CATALOG_SQL =
+      "ALTER TABLE " + CATALOG_TABLE_VIEW_NAME + " ADD COLUMN " + TYPE + " VARCHAR(5)";
+
+  private static final String GET_SQL =
       "SELECT * FROM "
-          + CATALOG_TABLE_NAME
+          + CATALOG_TABLE_VIEW_NAME
           + " WHERE "
           + CATALOG_NAME
           + " = ? AND "
           + TABLE_NAMESPACE
           + " = ? AND "
           + TABLE_NAME
-          + " = ? ";
-  static final String LIST_TABLES_SQL =
+          + " = ? AND "
+          + TYPE
+          + " = ?";
+  private static final String GET_TABLE_SQL =
+      GET_SQL
```

Comment:
maybe it would be better to have the full query here with `... AND (TYPE = ? OR TYPE IS NULL)...` so that it's clearer?

Reply:
Yeah, it would be probably cleaner. I will do that.